KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift

```diff
@@ -412,18 +413,11 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
     
     private func getPaymentNonceAccordingToAuthState() -> String? {
         switch Karhoo.configuration.authenticationMethod() {
-        case .tokenExchange(settings: _), .karhooUser: return retrievePaymentNonce()
+        case .tokenExchange(settings: _), .karhooUser: return view?.getPaymentNonce()?.nonce
```

Comment:
What's the difference between these cases and the default? Looks like we're returning the same thing in all cases.